File: C++/프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    unordered_map<string, int> clothmap;
    for (int i=0; i<clothes.size(); i++)
    {
        clothmap[clothes[i][1]]++;
    }
    
    int answer = 1;
    
    for (const auto& [key, value] : clothmap)
    {
        answer *= (value+1);
    }

    return answer-1;
}

File: Python3/프로그래머스/2/42578. 의상/의상.py
from collections import defaultdict

def solution(clothes):
    answer = 1
    d = defaultdict(list)
    
    for cloth in clothes :
        d[cloth[1]].append(cloth[0])
        
    for v in d.values() :
        answer *= (len(v)+1)
        
    answer -= 1
    
    return answer